app.js

```diff
@@ -253,6 +253,31 @@ app.get("/UserHome", function (req, res) {
     }
 });
 
+app.get("/UserHome", async function(req, res) {
+    let fullNames = {};
+    function getFullName(username){
+        User.findOne({
+            username: username
+        }, function(err, found) {
+            if(found) {
```

Comment:
combine all this to the existing function, in order to keep the `req.isAuthenticated()` check and `res.locals.username = req.user.username;` (it's used for displaying name on tab title).

Reply:
Done. Hopefully there isn't a problem with having multiple ```res.render()``` calls in the same function. 
Another question I have is if ```username: username``` on line 255 needs to be changed to ```username: req.user.username```?